### PR TITLE
feat: add type_ignore support to graph_view (#117)

### PR DESCRIPTION
**Pull request by Maqbool61**
## Summary

Closes #117

`flow_view` and `lenet_view` already supported `type_ignore` to skip certain
layer types from the diagram. This PR brings `graph_view` to full parity and
also adds the missing `plot_ignore_layers.py` example for the graph style.

## Problem

`graph_view` had no `type_ignore` parameter at all. Users who wanted to hide
noisy layers like `nn.ReLU`, `nn.BatchNorm2d`, or `nn.Flatten` from the graph
diagram had no way to do so, even though the exact same feature already worked
in the other two styles.

The fix mirrors what `flow_view` and `lenet_view` already do: filter the
columns before building the diagram, so ignored layer types never get a node
drawn for them and their edges are automatically dropped.

## Changes

**`render.py`**
- Added `type_ignore: list[type] | None = None` to `GraphStyleOptions`
- `_render_graph()` forwards it to `graph_view()`

**`graph.py`**
- Added `type_ignore: list | None = None` to `graph_view()` signature with
  matching docstring entry
- Applied the same two-line column filter used in `flow_view`/`lenet_view`
  before passing columns into `_create_architecture()`:
```python
  filtered_columns = [[layer for layer in column if type(layer.module) not in type_ignore] for column in raw_columns]
  filtered_columns = [column for column in filtered_columns if column]
```
- The existing edge guard (`edge[0] in id_to_node_list_map`) already drops
  dangling edges from ignored layers automatically — no extra work needed

**`tests/test_graph.py`**
- `test_graph_view_with_type_ignore` — smoke test that `type_ignore=[nn.ReLU]`
  runs without error
- `test_graph_view_type_ignore_reduces_diagram` — asserts the customized output
  actually differs in pixels from the default, confirming ignored layers were
  removed from the rendered diagram

**`docs/examples/graph/plot_ignore_layers.py`**
- New example matching the existing `flow/plot_ignore_layers.py` and
  `lenet_style/plot_ignore_layers_lenet_style.py`, using `style="graph"` with
  `type_ignore=[nn.ReLU, nn.Flatten]` and `show_input=False`

## Backward Compatibility

No breaking changes. `type_ignore=None` (the default) reproduces existing
behavior exactly — `None` is normalized to `[]` internally, so the filter
is a no-op when unset.

## Usage

```python
import torch.nn as nn
import visualtorch

model = nn.Sequential(
    nn.Conv2d(3, 16, kernel_size=3, padding=1),
    nn.ReLU(),
    nn.MaxPool2d(2, 2),
    nn.Linear(16, 10),
)

# Hide ReLU nodes from the graph diagram
img = visualtorch.render(
    model,
    input_shape=(1, 3, 224, 224),
    style="graph",
    type_ignore=[nn.ReLU],
)
```

### CODE REVIEW

**Review by willyfh — APPROVED** [review]
Thanks, this is clean. LGTM!
